### PR TITLE
Updated Thai localization formatting

### PR DESCRIPTION
**Pull request by whs**
This PR adds the missing Thai localization formatting settings.

### CODE REVIEW

**Comment by timgraham** [comment]
Could you please find a colleague to +1 the changes?

File: django/conf/locale/th/formats.py [hunk 1]
```diff
@@ -9,13 +9,25 @@
...
 
 # The *_INPUT_FORMATS strings use the Python strftime format syntax,
 # see http://docs.python.org/library/datetime.html#strftime-strptime-behavior
-# DATE_INPUT_FORMATS =
-# TIME_INPUT_FORMATS =
-# DATETIME_INPUT_FORMATS =
+DATE_INPUT_FORMATS = [
+    '%d/%m/%Y',  # 25/10/2006
+    '%d %b %Y',  # 25 ต.ค. 2006
+    '%d %B %Y',  # 25 ตุลาคม 2006
```
Comment (dtinth) [inline django/conf/locale/th/formats.py]: Is it possible to convert year type? (e.g. 2006 &rarr; 2549)

Comment (timgraham) [inline django/conf/locale/th/formats.py]: I'm not sure what this comment is suggesting to do. Change the year in the example?

Comment (dtinth) [inline django/conf/locale/th/formats.py]: In Thailand it’s customary to use the [Thai solar calendar](https://en.wikipedia.org/wiki/Thai_solar_calendar) system (as it’s the official legal calendar of Thailand).

Dates and months are the same as the common Gregorian Calendar, but years are in Buddhist Era instead of the Christian/Common Era. Just add 543 to the year number when displaying, and subtract 543 from the year number when parsing.

Comment (timgraham) [inline django/conf/locale/th/formats.py]: I'm not sure -- @claudep, any advice?

Comment (claudep) [inline django/conf/locale/th/formats.py]: The Python datetime module assumes a Gregorian calendar (https://docs.python.org/3.4/library/datetime.html#date-objects). We do not yet support other calendar systems in Django.

Comment (whs) [inline django/conf/locale/th/formats.py]: Thai calendar (Buddhist Era) is also a Gregorian calendar, but we start counting 543 years earlier than Common Era.

I think it wouldn't be hard to convert B.E. to C.E. A simple heuristic I used is that if the input year is over 2400 it should be B.E. (The current year is 2560) However, I think this is out of scope of this PR and implementing such system would require a lot more discussion.

Comment (claudep) [inline django/conf/locale/th/formats.py]: Sure, if you can provide a patch in another ticket, it would merit consideration.
